### PR TITLE
fix(gerrit): `commitBody` option not being effective

### PR DESCRIPTION
**Pull request by felipecrs**
## Changes

This fixes the `commitBody` setting in `renovate.json` being ignored/removed when Renovate needs to update the change subject on Gerrit.

The previous code to update the commit message with the PR title was overriding the whole commit message with the new title. Now, it takes into account the previous commit message and only override its subject.

## Context

I believe the issue is self-explanatory regarding the context.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by felipecrs** [comment]
@NiasSt90 if you can please take a look.

File: lib/modules/platform/gerrit/client.ts [hunk 1]
```diff
@@ -103,15 +104,17 @@ class GerritClient {
...
-    prTitle: string,
+    currentMessage: string,
+    newSubject: string,
   ): Promise<void> {
-    await this.setCommitMessage(
-      number,
-      `${prTitle}\n\nChange-Id: ${gerritChangeID}\n`,
+    // Replace first line of the commit message with the new subject
+    const newMessage = currentMessage.replace(
+      new RegExp(`^.*$`, 'm'),
```
Comment (viceice) [inline lib/modules/platform/gerrit/client.ts:114]:
```suggestion
      regEx(`^.*$`, 'm'),
```

use our regex helper

**Review by rarkins — APPROVED** [review]

File: lib/modules/platform/gerrit/types.ts [hunk 2]
```diff
@@ -43,21 +43,26 @@ export interface GerritChange {
   labels?: Record<string, GerritLabelInfo>;
   reviewers?: Record<GerritReviewersType, GerritAccountInfo[]>;
   messages?: GerritChangeMessageInfo[];
-  current_revision?: string;
+  current_revision: string;
```
Comment (viceice) [inline lib/modules/platform/gerrit/types.ts:46]: Are you sure those new / updated properties are always there?

Comment (felipecrs) [reply lib/modules/platform/gerrit/types.ts:46]: Yes, all the search API used to populate these data is requesting such information through the `o=` query. Therefore, Gerrit API will always provide them.

**Pull request merged**